transport/stream.go

```diff
@@ -9,7 +9,8 @@ import (
 )
 
 const (
-	maxStreamID = math.MaxInt16
+	maxStreamID   = math.MaxInt16
+	eventStreamID = math.MaxUint16 // TODO: is this correct?
```

Comment:
It's seems utterly wrong

```
  A frame has a stream id (a [short] value). When sending request messages, this
  stream id must be set by the client to a non-negative value (negative stream id
  are reserved for streams initiated by the server; currently all EVENT messages
  (section 4.2.6) have a streamId of -1). If a client sends a request message
  with the stream id X, it is guaranteed that the stream id of the response to
  that message will be X.
```

- stream ID is 2 bytes singed int i.e. Int16
- EVENTS use negative IDs

Reply:
Apparently `ffff` is binary coding of both `-1` as int16 and `math.MaxUint16`...

Can we use int16(-1)?

Reply:
```
To describe the layout of the frame body for the messages in Section 4, we define the following:
    [int]            A 4 bytes integer
    [long]         A 8 bytes integer
    [short]        A 2 bytes unsigned integer
    ...
```
Since here short is described as unsigned, we were hesitating with the type. In frame we have:
```
// StreamID is a type alias for Short.
type StreamID = Short
```
and in transport we have:
```
type connReader struct {
	...
	h      map[frame.StreamID]responseHandler
        ...
}
```
So if we use **int16(-1)** we would still require casting it to **uint16** in other parts of code.

Can we stay with **math.MaxUint16** and mark it with appropriate comment?
Or should we change the rest of the code so that StreamID is of type **int16**, because perhaps that's a little bug in **gocql binary protocol**?

BTW:
`maxStreamID = math.MaxInt16`
So it is probably doable without greater changes.

Reply:
This is very interesting!
[Here]https://github.com/apache/cassandra/blob/2b2c6decfafc6235ad537e72073fab2fd4467e2f/doc/native_protocol_v4.spec#L160) they clearly state that those 2 bytes
- can can hold up to 0x800 (max int 16 singed)
- -1 is EVENTS stream

I think that we can change 

```
// StreamID is a type alias for Short.
type StreamID = int16

// Header spec https://github.com/apache/cassandra/blob/trunk/doc/native_protocol_v4.spec#L101.
type Header struct {
...
	StreamID StreamID
}
```